executor/oci/resolvconf.go

```diff
@@ -51,6 +51,22 @@ func GetResolvConf(ctx context.Context, stateDir string, idmap *idtools.Identity
 				} else {
 					if fi.ModTime().Before(fiMain.ModTime()) {
 						generate = true
+					} else {
+						var oriFileHasLocalDNS bool
```

Comment:
This does not look safe if this function is called in parallel with different `hostNetMode` value. I think the only way would be to return a different path for these cases.

There is also this issue that is related https://github.com/moby/buildkit/issues/3096 (but you don't have to fix this as part of this PR)